### ISSUE TITLE
SwiftSlash does not work in Amazon Linux docker

### ISSUE THREAD

**Issue by tayloraswift** [tannerdsilva/SwiftSlash#7]
i only get `Error: badAccess` whenever i try to launch any command. neither the sync nor AsyncStream-based APIs work. 

test code: 

```swift 
@preconcurrency import SwiftSlash

public  
func _lastModified(_ path:String) async throws 
{
    let command:Command = .init(execute: "git", 
        arguments: ["log", "-n", "1", "--format='%H'", path])
    let git:ProcessInterface = .init(command: command, stdin: .nullPipe)
    
    try await git.launch()
    print("launched")
    for await data:_ in await git.stdout 
    {
        print(data)
    }
    print("completed")
}
```

dockerfile: 

```
FROM amazonlinux:latest
WORKDIR /home/ec2-user/

RUN yum -y update 
RUN yum -y install lsb-release curl git cmake ninja-build clang python uuid-dev libicu-dev icu-devtools libbsd-dev libedit-dev libxml2-dev libsqlite3-dev swig libpython-dev libncurses5-dev pkg-config tar

ENV SWIFTENV_ROOT /home/ec2-user/.swiftenv
RUN mkdir -p $SWIFTENV_ROOT
ADD https://github.com/kylef/swiftenv/archive/1.2.0.tar.gz /tmp/swiftenv.tar.gz
RUN tar -xzf /tmp/swiftenv.tar.gz -C $SWIFTENV_ROOT --strip 1

ENV PATH $SWIFTENV_ROOT/bin:$SWIFTENV_ROOT/shims:$PATH
ENV SWIFT_VERSION DEVELOPMENT-SNAPSHOT-2022-02-03-a
RUN swiftenv install https://download.swift.org/development/amazonlinux2/swift-$SWIFT_VERSION/swift-$SWIFT_VERSION-amazonlinux2.tar.gz
RUN swiftenv global $SWIFT_VERSION

# enables showing the PWD prompt
RUN echo "PS1='\w\$ '" >> .bashrc
RUN echo "force_color_prompt=yes" >> .bashrc
ENV TERM xterm-256color

ENTRYPOINT /bin/bash --init-file .bashrc
```

**Comment by tannerdsilva** [comment]
Hey, thanks for the feedback @kelvin13!

I saw this issue and initially thought it was a bug, and began digging in. I quickly realized that we actually don't have an explicit bug here, but rather, a lack of clarity on the Command initializer functions. I will try rectify this confusion with a new tag `3.4.0`.

Until this new version is ready to push, there are simple changes that can be implemented in your project to prevent this error from being thrown.

### Problem
The problem you are hitting is that the `Command` initializer you are calling expects `execute:String` to be an absolute system path, not a relative executable name. I vaguely alluded to this in the functions documentation. Even as the creator of this framework, **it was not obvious to me that you were misusing the `Command.init` function when I first saw your code.**

### Solution [current release 3.3.x]
I would resolve the explicit path for `git` (or any other command) by first running something like this:
```
let gitPath = String(data:try await Command(bash:"which git").runSync().stdout.first, encoding:.utf8)
``` 
...and passing the resulting string into a `Command.init(execute:...)`

### Solution [future release 3.4.0]
I believe SwiftSlash needs to offer a Command initializer that can take a relative system command as you have showcased here. I believe it is perfectly reasonable to simply pass "git" into a command initializer and expect it to resolve properly based on the current execution environment. Furthermore, I think the specific error that is being thrown can be more descriptive, making it more clear to the user that path resolution has failed for the executable target.

I will be leaving this issue open until tag `3.4.0` can be pushed into the `master` branch.

**Comment by tayloraswift** [comment]
thank you for the quick response!

**Comment by tannerdsilva** [comment]
I am sincerely sorry it took me so long to get this release out. The branch where I made these developments found no end and evolved into the next major release.

As promised above, v3.4 has been released. More info found [here](https://github.com/tannerdsilva/SwiftSlash/pull/11#issue-1600240553) and in the [documentation](https://swiftslash.com/documentation/swiftslash/command) for the revised Command struct.
